homeassistant/components/home_connect/select.py

```diff
@@ -148,3 +224,57 @@ async def async_select_option(self, option: str) -> None:
                     SVE_TRANSLATION_PLACEHOLDER_PROGRAM: program_key.value,
                 },
             ) from err
+
+
+class HomeConnectSelectEntity(HomeConnectEntity, SelectEntity):
+    """Select setting class for Home Connect."""
+
+    entity_description: HomeConnectSelectEntityDescription
+
+    def __init__(
+        self,
+        coordinator: HomeConnectCoordinator,
+        appliance: HomeConnectApplianceData,
+        desc: HomeConnectSelectEntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        setting = appliance.settings.get(cast(SettingKey, desc.key))
+        if setting and setting.constraints and setting.constraints.allowed_values:
+            self._attr_options = [
+                desc.values_translation_key[option]
+                for option in setting.constraints.allowed_values
+                if option in desc.values_translation_key
+            ]
+        super().__init__(
```

Comment:
Generally, it's good practice so call `super().__init__` first and then set the child attributes unless there's a reason to do otherwise, since the child inherits the parent and the child may want to override parent attributes.

Reply:
Hmmm
Back on day I tried to do it that way, and some issues appeared. I just tried again and tests works. Pushing it then😅